it/uk/gov/hmrc/transitmovements/repositories/DeparturesRepositorySpec.scala

```diff
@@ -129,78 +125,119 @@ class DeparturesRepositorySpec
+            .sortBy(_.received)
+            .reverse
+        )
         .value
     val departure = arbitrary[Departure].sample.value.copy(messages = messages)
 
     await(repository.insert(departure).value)
 
-    val result = await(repository.getDepartureMessageIds(departure.enrollmentEORINumber, departure._id).value)
+    val result = await(repository.getDepartureMessageIds(departure.enrollmentEORINumber, departure._id, None).value)
     result.right.get.get should be(
       departure.messages.map(
         x => x.id
       )
     )
   }
 
+  "getDepartureMessageIds" should "return messageIds if there are messages that were received since the given time" in {
+    val dateTime = arbitrary[OffsetDateTime].sample.value
+
+    val messages =
+      NonEmptyList
+        .fromList(
+          List(
+            arbitraryMessage.arbitrary.sample.value.copy(received = dateTime.plusMinutes(1)),
+            arbitraryMessage.arbitrary.sample.value.copy(received = dateTime),
+            arbitraryMessage.arbitrary.sample.value.copy(received = dateTime.minusMinutes(1))
+          )
+        )
+        .value
+
+    val departure = arbitrary[Departure].sample.value.copy(messages = messages)
+
+    await(repository.insert(departure).value)
+
+    val result = await(repository.getDepartureMessageIds(departure.enrollmentEORINumber, departure._id, Some(dateTime)).value)
+    result.right.get.get should be(
+      NonEmptyList.fromListUnsafe(
+        departure.messages
+          .take(2)
+          .map(
+            x => x.id
+          )
+      )
+    )
+  }
+
   "getDepartureMessageIds" should "return none if the departure doesn't exist" in {
-    val result = await(repository.getDepartureMessageIds(EORINumber("ABC"), DepartureId("XYZ")).value)
+    val result = await(repository.getDepartureMessageIds(EORINumber("ABC"), DepartureId("XYZ"), None).value)
     result.right.get.isEmpty should be(true)
   }
 
   "getDepartureIds" should
     "return a list of departure ids for the supplied EORI sorted by last updated, latest first" in {
-      val result = await(repository.getDepartureIds(dbSetup.eoriGB).value)
+      GetDeparturesSetup.setup()
+      val result = await(repository.getDepartureIds(GetDeparturesSetup.eoriGB).value)
 
       result.right.get.value should be(NonEmptyList(DepartureId("10004"), List(DepartureId("10001"))))
     }
 
   it should "return no departure ids for an EORI that doesn't exist" in {
+    GetDeparturesSetup.setup()
     val result = await(repository.getDepartureIds(EORINumber("FR999")).value)
 
     result.right.get should be(None)
   }
 
   it should "return no departure ids when the db is empty" in {
-    await(repository.collection.drop().toFuture())
+    // the collection is empty at this point due to DefaultPlayMongoRepositorySupport
     val result = await(repository.getDepartureIds(EORINumber("FR999")).value)
     Thread.sleep(10000)
```

Comment:
I'm being picky here, but sleeping for 10 seconds in a test might not be the best thing if it can be avoided

Reply:
I didn't spot that -- yeah, that needs to be looked at.

Reply:
Yeah, I realise its outside your PR, but thought it was worth mentioning

Reply:
That's an error  that was left in.
Can be removed - the await catches it.

Reply:
Yup, I'll ditch it now.